### PR TITLE
worker warmup callback

### PR DESCRIPTION
**Pull request by divaykin**
feature allowing to run certain code in the beginning of the worker thread.
one can use it to set thread priority, affinity mask, etc.

### CODE REVIEW

File: include/spdlog/spdlog.h [hunk 1]
```diff
@@ -63,7 +63,7 @@ void set_level(level::level_enum log_level);
 
 // Turn on async mode and set the queue size for each async_logger
 
-void set_async_mode(size_t queue_size);
+void set_async_mode(size_t queue_size, const std::function<void()>& worker_warmup_cb);
```
Comment (xaqq) [inline include/spdlog/spdlog.h]: Shouldn't this use a default value of `nullptr` ?

**Comment by xaqq** [comment]
This looks like a nice addition.
See my question in diff comment.
Also there is a few indentation problem.

**Comment by gabime** [comment]
Yes. I agree. This should have default null

**Comment by divaykin** [comment]
No problem. For indentation - do you use always spaces? I typically use tabs

**Comment by gabime** [comment]
You can run the top level astyle.sh

**Comment by divaykin** [comment]
Should be better now. astyle.sh does some extra mods which I guess you don't really want - don't know why exactly, no expert

**Pull request merged**